client/src/extension.ts

```diff
@@ -235,6 +235,17 @@ export function activate(context: ExtensionContext) {
   // Start the client. This will also launch the server
   client.start();
 
+  // Restart the language client automatically certain configuration changes.
+  // These are typically settings that affect the capabilities of the language
+  // client, and because of that requires a full restart.
+  context.subscriptions.push(
+    workspace.onDidChangeConfiguration(({ affectsConfiguration }) => {
+      if (affectsConfiguration("rescript.settings.inlayHints")) {

```

Comment:
Is there a special reason to restart only when this specific setting changes?
I imagine as a user I would be quite confused "I think last time I could just change setting, this time I need to restart, strange...".

Reply:
It's only restarted automatically when required. Some configs can be changed on the fly, but other configs require restarting the server because they affect how the lsp is instantiated. If we don't have this then the user would need to be promoted to restart the server manually to have their config take effect.

Reply:
The question is why not the other changes too.
How do I know other configs don't need restarting the server?
And if they don't need restating the server today, how do I know they don't need to restart the server tomorrow.
And if tomorrow they will need a restart, what are the chances the person doing the change will think of that, realise this piece of code exists, and come here to update it?

Overall, why special case thins config, is the question.